libqtile/group.py

```diff
@@ -285,20 +292,29 @@ def mark_floating(self, win, floating):
                 # already floating
                 pass
             else:
+                # Remove from the tiled windows list if the window is not fullscreen
+                if not win.fullscreen:
+                    self.tiled_windows.remove(win)
                 for i in self.layouts:
-                    i.remove(win)
-                    if win is self.current_window:
-                        i.blur()
+                    # Remove the window from the layout if it is not fullscreen
+                    if not win.fullscreen:
+                        i.remove(win)
+                        if win is self.current_window:
+                            i.blur()
                 self.floating_layout.add(win)
                 if win is self.current_window:
                     self.floating_layout.focus(win)
         else:
             self.floating_layout.remove(win)
             self.floating_layout.blur()
             for i in self.layouts:
-                i.add(win)
+                # A window that was fullscreen should only be added if it was not a tiled window
+                if win not in self.tiled_windows:
+                    i.add(win)
                 if win is self.current_window:
                     i.focus(win)
```

Comment:
We can save more checks here too (especially as `win not in self.tiled_windows` is O(n)):
```suggestion
            # A window that was fullscreen should only be added if it was not a tiled window
            if win not in self.tiled_windows:
		for i in self.layouts:
                    i.add(win)
	    if win is self.current_window:
		for i in self.layouts:
                    i.focus(win)
```

Reply:
Also makes sense, this is much better